library/pem.c

```diff
@@ -240,6 +240,29 @@ static int pem_aes_decrypt(unsigned char aes_iv[16], unsigned int keylen,
 }
 #endif /* MBEDTLS_AES_C */
 
+#if defined(MBEDTLS_DES_C) || defined(MBEDTLS_AES_C)
+static int pem_check_pkcs_padding(unsigned char *input, size_t input_len, size_t *data_len)
+{
+    /* input_len > 0 is guaranteed by mbedtls_pem_read_buffer(). */
+    size_t pad_len = input[input_len - 1];
+    size_t i;
+
+    if (pad_len > input_len) {
+        return MBEDTLS_ERR_PEM_BAD_INPUT_DATA;
```

Comment:
I would expect this function to return `MBEDTLS_ERR_PEM_PASSWORD_MISMATCH` on error (both cases). That's what that error is for.